include/LightGBM/utils/common.h

```diff
@@ -26,6 +27,10 @@
 #include <utility>
 #include <vector>
 
+#define FMT_HEADER_ONLY
```

Comment:
oh!! If we're just doing this, then can you please revert the changes in `R-package/Makevars.in` and `R-package/Makevars.win.in`?

We don't need to be passing `-DFMT_HEADER_ONLY` at compile time if you'r just defining it right here.

```text
/opt/csw/bin/g++-5.5 -std=gnu++11 -I"/opt/csw/lib/R/include" -DNDEBUG -I./include -DFMT_HEADER_ONLY -DMM_PREFETCH=1 -DMM_MALLOC=1 -DUSE_SOCKET -DLGB_R_BUILD  -I/opt/csw/include  -pthread -fPIC  -O2 -pipe -fno-omit-frame-pointer -m32 -march=pentiumpro  -c application/application.cpp -o application/application.o
In file included from ./include/LightGBM/config.h:16:0,
                 from ./include/LightGBM/application.h:8,
                 from application/application.cpp:5:
./include/LightGBM/utils/common.h:30:0: warning: "FMT_HEADER_ONLY" redefined
 #define FMT_HEADER_ONLY
 ^
<command-line>:0:0: note: this is the location of the previous definition
```